### PR TITLE
Use multiple dispatch for `plot` functions

### LINKED ISSUE #152: Make all plotting functions overload `plot` instead of having different names
*Except the function `animate_evolution`*.

I don't know why I wasn't using multiple dispatch in the first place! So in short, rename all `plot_obstacle`, `plot_particle`... etc. to just `plot`, and `import plot` from whatever plotting package we are using.

### PR DESCRIPTION
**Pull request by AnAverageHuman**
Don't merge this yet. Fixes #152.

Tests are passing, but `plot` is not defined.  As far as I understand, to extend `PyPlot.plot()`, we need something like `import PyPlot: plot`, but since we don't explicitly REQUIRE PyPlot, that's not safe.  Is there a better way to handle that?

### CODE REVIEW

**Comment by Datseris** [comment]
You bring up a good point.

Here is the thing. Does it _actually_ work to do `import PyPlot: plot` ? We don't REQUIRE `PyPlot` in the standard require file, however we do require it through the usage of the `Requires` module. 

If `import PyPlot: plot` works and one can extend methods then this is fine and safe. Because
```
using Requires
function __init__()
    @require PyPlot="d330b81b-6aea-500a-939a-2ce795aea3ee" begin
        dir = joinpath(@__DIR__, "plotting")
        for f in readdir(dir)
            include(joinpath(dir, f))
        end
    end
end
```
ensures that this code is only loaded when PyPlot is _already_ in scope.

File: docs/src/basic/phasespaces.md [hunk 1]
```diff
@@ -30,11 +30,11 @@ t = 200 # how long to evolve each one
 
 ξς, φς, ις = boundarymap(bd, t, n)
 
-using PyPlot # enables plot_boundarymap function
+using PyPlot # enables plot function
 
 colors = ["C$(rand(1:9))" for i in 1:n] # random colors
 
-plot_boundarymap(ξς, φς, ις, color = colors)
+plot(ξς, φς, ις, color = colors)
```
Comment (Datseris) [inline docs/src/basic/phasespaces.md]: Could you please revert the change of `plot_boundarymap`? This function can't be safely extend `plot` because it uses vectors of numbers.

File: src/plotting/particles.jl [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
 
 """
-    plot_cyclotron(p::MagneticParticle; use_cell=true, kwargs...)
+    plot(p::MagneticParticle; use_cell=true, kwargs...)
 Plot the circle traced by the free particle motion. Optionally use `p.current_cell` for
 the particle's position. The user provided `kwargs...`
 are passed onto `PyPlot.plt[:Circle]`.
 """
-function plot_cyclotron(p::MagneticParticle; use_cell=true, kwargs...)
+function plot(p::MagneticParticle; use_cell=true, kwargs...)
```
Comment (Datseris) [inline src/plotting/particles.jl]: It is now ambiguous when you call `plot(p::MagneticParticle)` of what should happen. Should it plot the cyclotron method, or should it plot the particle method (arrow and point) ?

Comment (AnAverageHuman) [reply src/plotting/particles.jl]: Reverting this one for now.

**Comment by AnAverageHuman** [comment]
It seems to work now, except for an ugly warning when `using DynamicalBilliards`:

```
┌ Warning: Package DynamicalBilliards does not have PyPlot in its dependencies:
│ - If you have DynamicalBilliards checked out for development and have
│   added PyPlot as a dependency but haven't updated your primary
│   environment's manifest file, try `Pkg.resolve()`.
│ - Otherwise you may need to report an issue with DynamicalBilliards
└ Loading PyPlot into DynamicalBilliards from project dependency, future warnings for DynamicalBilliards are suppressed.
```

Documenter on Travis is also throwing `!! Duplicate docs found for 'plot'. [src/visualizing.md]`; not sure if that's an error or a warning.

File: docs/src/visualizing.md [hunk 3]
```diff
@@ -2,10 +2,10 @@ All plotting functionality of `DynamicalBilliards` lies within a few well-define
...
 ```@docs
-plot_obstacle
-plot_particle
+plot
+plot
 plot_cyclotron
-plot_billiard
+plot
 animate_evolution
 ```
```
Comment (Datseris) [inline docs/src/visualizing.md]: Here is the problem for the warning you are seeing: Remove all the extra plots (leave only one line with `plot` and instead of just `plot` do instead `DynamicalBilliards.plot`.

**Comment by AnAverageHuman** [comment]
Rebased with master.  Logs on CI look okay.

**Comment by Datseris** [comment]
We got a problem: https://github.com/MikeInnes/Requires.jl/issues/58

I checked out your branch to test it, and it seems like this does not work, as it says:

```
┌ Warning: Package DynamicalBilliards does not have PyPlot in its dependencies:│ - If you have DynamicalBilliards checked out for development and have│   added PyPlot as a dependency but haven't updated your primary│   environment's manifest file, try `Pkg.resolve()`.│ - Otherwise you may need to report an issue with DynamicalBilliards└ Loading PyPlot into DynamicalBilliards from project dependency, future warnings for DynamicalBilliards are suppressed.
```
Let's hope this is fixable...

**Comment by Datseris** [comment]
@AnAverageHuman thank you very much for your contribution!

Do you also use `DynamicalBilliards` ? Out of curiosity
